doc/source/user/tutorial-svd.rst

```diff
@@ -0,0 +1,468 @@
+individual image (otherwise, `matplotlib` will default to a colormap that does
+not correspond to the real data).
+
+In our case, we are approximating the blue portion of the image, so we will use
+the colormap ``Blues``:
+
+::
+
+    >>> plt.imshow(approx_blue, cmap=plt.cm.Blues)
+
+Now, you can go ahead and repeat this experiment with other values of `k`, and
+each of those should give you a slightly better (or worse) image depending on
+the value you choose.
+
+Just for good measure, to see how the original blue array looks like, you can
+do
+
+::
+
+    >>> plt.imshow(blue_array, cmap=plt.cm.Blues)
+
+Applying to all colors
+----------------------
+
+Now, we want to do the same kind of operation, but to all three colors. Our
+first instinct might be to repeat the same operation we did above to each color
+matrix individually. However, numpy's `broadcasting` takes care of this
+for us.
+
+If our array has more than two dimensions, then the SVD can be applied in
+“stacked” mode. However, the linear algebra functions in numpy expect to see a
+stacking of the form ``(N, M, M)``, where the first axis represents the number
+of stacked matrices (i.e. ``(N, M, M)`` is interpreted as a “stack” of ``N``
+matrices, each of size ``(M, M)``.)
+
+In our case,
+
+::
+
+    >>> img_array.shape
+    (768, 1024, 3)
+
+so we need to permutate the axis on this array to make get a shape like
+``(3, 768, 1024)``. Fortunately, the `numpy.transpose` function can do that for
+us:
+
+``np.transpose(x, axes=(i, j, k))`` 
+
+indicates that the axis will be reordered such that the final shape of the
+transposed array will be reordered according to the indices ``(i, j, k)``.
+
+Let's see how this goes for our array:
+
+::
+
+    >>> img_array_transposed = np.transpose(img_array, (2, 0, 1))
+    >>> img_array.shape
+    (3, 768, 1024)
+
+Now we are ready to apply the SVD:
+
+::
+
+    >>> U, s, Vt = LA.svd(img_array_transposed)
+
+Finally, to obtain the full approximated image, we need to reassemble these
+matrices into the approximation. Now, note that
+
+::
+
+    >>> U.shape, s.shape, Vt.shape
+    ((3, 768, 768), (3, 768), (3, 1024, 1024))
+
+To build the final approximation matrix, we must understand how multiplication
+across different axes works.
+
+Products with n-dimensional arrays
+----------------------------------
+
+If you have worked before with only one- or two-dimensional arrays in NumPy,
+you might use `numpy.dot` and `numpy.matmul` (or the ``@`` operator)
+interchangeably. However, for n-dimensional arrays, they work in very different
+ways. For more details, check the documentation `numpy.matmul`.
+
+Now, to build our approximation, we need first to make sure that our singular
+values are ready for multiplication, so we build our ``Sigma`` matrix similarly
+to what we did before. The ``Sigma`` array must have dimensions
+``(3, 768, 1024)``.
+
+::
+
+    >>> Sigma = np.zeros((3, 768, 1024))
```

Comment:
This does not seem like the correct way to generate this array.

But, also, it does not seem desirable to make such a large Sigma.  Is that necessary?  Can't broadcasting do some of the work here?

At this point, I feel like we may be overwhelming the audience a bit.  While it is true that with broadcasting and other tricks you can turn everything into a near one-liner, something like `apply_along_axis` would probably be about as effective and less confusing to beginners.

Maybe stop the tutorial before the color section?

Reply:
If we did my suggestion in https://github.com/numpy/numpy/pull/15295#discussion_r364660316, then this would become something like:

```
Sigma = np.apply_along_axis(np.diag, s, -1)
```
Things get awkward because the singular value matrix is rectangular, which isn't useful for compression anyway.

Reply:
I felt that adding `np.apply_along_axis` would be more complicated to explain. I was trying to keep concepts as close to linear algebra as possible, and while the transpose is a well known operation to most people who use linear algebra in their research/work, `apply_along_axis` might be a little more tricky to explain. 

About the length of the tutorial, I could break this tutorial into 2 shorter ones; one about the more basic operations and broadcasting - done only on a gray scale imge - and another one about multiple axes. However, i felt that the Absolute Beginner's tutorial will work with the basic operations so I wanted to take this one a little farther. 

Let me know if you think it would be better to break this one, and I can reformulate it. Thanks!

Reply:
I agree with keeping things simple here. `zeros` is trivial to understand, `apply_along_axis(...)` isn't.

And also +1 on keeping scope as is, it doesn't really make sense to split in two since then the non-linalg stuff is covered by other parts of the docs already. The alternative that could be considered is shortening, but the current length doesn't bother me.

Reply:
The simplicity argument is a strong one.  At the same time, I think it is important that we teach the mechanisms we would use ourselves.  Habits are surprisingly hard to un-teach!  You'll almost never see double for loops in practice and besides that you most often do not need large arrays to represent repeating elements (given tools such as broadcasting, etc.). But, let's see how it all comes together; I'm happy to do another round of review after a round of editing.

Reply:
Ah, I was just commenting on the `np.zeros` usage. Agree that the double for-loop below is undesired.

Reply:
I tried using `np.fill_diagonal` to make this better, let me know what you all think.

Reply:
looks good to me now